praw/models/reddit/emoji.py

```diff
@@ -0,0 +1,81 @@
+"""Provide the EMOJI class."""
+from ...const import API_PATH
+from ..listing.generator import ListingGenerator
+from .base import RedditBase
+from .redditor import Redditor
+
+
+class Emoji(RedditBase):
+    """An individual Emoji object."""
+
+    __hash__ = RedditBase.__hash__
+
+    @property
+    def mod(self):
+        """Provide an instance of :class:`.WikiPageModeration`."""
+        if self._mod is None:
+            self._mod = EmojiModeration(self)
+        return self._mod
+
+    def __init__(self, reddit, subreddit, name, revision=None, _data=None):
+        """Construct an instance of the WikiPage object.
+
+        :param revision: A specific revision ID to fetch. By default, fetches
+            the most recent revision.
+
+        """
+        self.name = name
+        self.subreddit = subreddit
+        super(Emoji, self).__init__(reddit, _data)
+        self._mod = None
+
+
+class EmojiModeration(object):
+    """Provides a set of moderation functions for a WikiPage."""
+
+    def __init__(self, wikipage):
+        """Create a WikiPageModeration instance.
+
+        :param wikipage: The wikipage to moderate.
+
+        """
+        self.emoji = emoji
+
+    def add(self, emoji_name, filepath):
+        """Add an emoji to this subreddit.
+
+        :param redditor: An emoji name (e.g., ``'cake'``) or
+            :class:`~.Emoji` instance.
+
+        To add ``'cake'`` as an editor on the wikipage ``'praw_test'`` try:
+
+        .. code:: python
+
+           reddit.subreddit('test').wiki['praw_test'].mod.add('cake')
+
+        """
+        data = {'filepath': filepath, 'mimetype': 'image/png'}
```

Comment:
How is the file actually uploaded?

Reply:
My best understanding here is that filepath would point to a local file path, and that the emoji_lease API would create the S3 key for it. This may be incorrect. Once we get the structure working, I think an image validation step within PRAW itself would be appropriate.

Reply:
[you're correct](https://www.reddit.com/r/redditdev/comments/7zbfzk/does_anyone_know_how_to_use_the_new_emoji/)